tests/publiccloud/sles4sap.pm

```diff
@@ -165,6 +165,78 @@ sub fence_node {
     }
 }
 
+=head2 workaround_bsc_1179529
+
+    workaround_bsc_1179529();
+
+Workaround bsc#1179529.
+=cut
+sub workaround_bsc_1179529 {
+    my ($self, %args) = @_;
+    $args{hostname} //= $self->{my_instance}->{instance_id};
+
+    record_soft_failure 'bsc#1179529 - [ha-sap-terraform-deployments_v6] DRBD resource fails after reboot on GCE';
+
+    # Get the UUID of DRBD device
+    my $drbd_conf_file = '/etc/drbd.d/sapdata.res';
+    my $drbd_device = $self->run_cmd(cmd => "awk '/[[:blank:]]+disk[[:blank:]]+\\/dev\\// { print substr(\$NF, 6, length(\$NF)-6) }' $drbd_conf_file", quiet => 1);
+    die "DRBD device can't be found!" unless defined $drbd_device;
+    my $drbd_id_device = $self->run_cmd(cmd => "ls -l /dev/disk/by-id/ 2>/dev/null | awk '/\\/${drbd_device}\$/ { print \$9 }' | head -n1", quiet => 1);
+    die "DRBD device ID can't be found!" unless defined $drbd_id_device;
+
+    # All node should be configured
+    my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1);
+    foreach my $node (@nodes) {
+        # Replace current device name by its ID in config file and sync cluster conf
+        $drbd_id_device =~ s/-0-part/-1-part/ unless $node eq $args{hostname};
+        $self->run_cmd(cmd => "sed -i '/[[:blank:]]*disk[[:blank:]]*\\/dev\\//d' $drbd_conf_file",               quiet => 1) if $node eq $args{hostname};
+        $self->run_cmd(cmd => "sed -i '/on $node {/a\\disk /dev/disk/by-id/${drbd_id_device};' $drbd_conf_file", quiet => 1);
+    }
+
+    # Sync config and restart DRBD resource
+    $self->run_cmd(cmd => 'csync2 -xF', quiet => 1);
+    $self->run_cmd(cmd => 'crm resource restart drbd-sapdata', timeout => 120, quiet => 1);
+}
+
+=head2 workaround_bsc_1179838
+
+    workaround_bsc_1179838();
+
+Workaround bsc#1179838.
+=cut
+sub workaround_bsc_1179838 {
+    my ($self) = @_;
+
+    unless ($self->run_cmd(cmd => 'systemctl --no-pager status pacemaker | grep -iq \'Active: inactive (dead)\'', rc_only => 1)) {
+        record_soft_failure 'bsc#1179838 - [ha-sap-terraform-deployments_v6] Pacemaker doesn\'t start correctly after a STONITH';
+        $self->run_cmd(cmd => 'systemctl --no-pager restart pacemaker', quiet => 1);
+        sleep 30;    # We need to wait a "little" before
+    }
+}
+
+=head2 workaround_bsc_1182701
+
+    workaround_bsc_1182701();
+
+Workaround bsc#1182701.
+=cut
+sub workaround_bsc_1182701 {
+    my ($self) = @_;
+
+    # All node should be configured
+    my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1);
+    foreach my $node (@nodes) {
+        # Get the STONITH resource
+        my $stonith_rsc = $self->run_cmd(cmd => "crm configure show | awk '/^primitive[[:blank:]]+rsc_gcp_stonith_.*_$node/ { print \$2 }'", quiet => 1);
+        # Check if workaround is already applied
+        if ($self->run_cmd(cmd => "crm resource param $stonith_rsc show method'", rc_only => 1)) {
```

Comment:
It seems like there's an extra `'` at the end of the command.

Reply:
Thx! Fixed in the next push.